cmd/valet_link.go

```diff
@@ -0,0 +1,90 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	"strings"
+
+	"github.com/mitchellh/cli"
+	"trellis-cli/trellis"
+)
+
+type ValetLinkCommand struct {
+	UI      cli.Ui
+	Trellis *trellis.Trellis
+}
+
+func (c *ValetLinkCommand) Run(args []string) int {
+	if err := c.Trellis.LoadProject(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	var environment string
+
+	switch len(args) {
+	case 0:
+		environment = "development"
+	case 1:
+		environment = args[0]
+	default:
+		c.UI.Error(fmt.Sprintf("Error: too many arguments (expected 0 or 1, got %d)\n", len(args)))
+		c.UI.Output(c.Help())
+		return 1
+	}
+
+	config, ok := c.Trellis.Environments[environment]
+	if !ok {
+		c.UI.Error(fmt.Sprintf("Error: %s is not a valid environment", environment))
+		return 1
+	}
+
+	c.UI.Info(fmt.Sprintf("Linking environment %s...\n", environment))
+
+	for key, site := range config.WordPressSites {
+		c.UI.Info(fmt.Sprintf("Linking site %s...", key))
+
+		canonical, _ := c.Trellis.HostsFromDomain(site.SiteHosts[0].Canonical, environment)
+		app := strings.TrimSuffix(canonical.String(), "." + canonical.TLD)
+
+		isSiteSslEnabled := site.Ssl["enabled"] == true
+
+		valetLink := execCommand("valet", "link", app)
```

Comment:
Could do this as well:

```go
valetArgs := []string{"link"}

if isSiteSslEnabled {
  valetArgs = append(valetArgs, "--secure")
}

valetLink := execCommand("valet", ...valetArgs, app)
```

Reply:
`valetLink := execCommand("valet", ...valetArgs, app)` gives syntax error

Have to do:
```go
valetArgs = append(valetArgs, app)
valetLink := execCommand("valet", valetArgs...)
```

Reply:
🤦‍♂ im still new to Go. `execCommand("valet", valetArgs..., app)` should be valid?

Reply:
Also syntax error.
Seems `...` must go last